### PR TITLE
Nex 1019/fix dbal default type

### PR DESCRIPTION
**Pull request by siwane**
This PR aims to fix bug introduces with dbal library: `"doctrine/dbal": "2.10.1"`.
It is casting string by default if no value has provided. In spanner context, the strong typing avoid to abuse of default value.
In this PR you will find explicit parameter casting for triple insertion and related unit tests

### CODE REVIEW

File: core/kernel/persistence/newsql/NewSqlRdf.php [hunk 1]
```diff
@@ -65,14 +70,39 @@ public function add(core_kernel_classes_Triple $triple)
...
+            ParameterType::STRING,
+            // object
+            ParameterType::STRING,
+            // l_language
+            ParameterType::STRING,
+            // author
+            ParameterType::STRING,
+            // epoch
+            ParameterType::STRING,
+        ];
```
Comment (gabrielfs7) [inline core/kernel/persistence/newsql/NewSqlRdf.php:106]: We can avoid maintaining the column schema on both places for other columns:

```suggestion
        return array_merge(
            [
                // id
                ParameterType::STRING
            ], 
            parent::getTripleParameterTypes()
        );
```

File: core/kernel/persistence/smoothsql/class.SmoothRdf.php [hunk 2]
```diff
@@ -68,7 +67,7 @@ public function add(core_kernel_classes_Triple $triple)
         $query = "INSERT INTO statements ( modelId, subject, predicate, object, l_language, epoch, author) "
             . "VALUES ( ? , ? , ? , ? , ? , ?, ?);";
 
-        $success = $this->getPersistence()->exec(
+        return $this->getPersistence()->exec(
```
Comment (gabrielfs7) [inline core/kernel/persistence/smoothsql/class.SmoothRdf.php:70]: nice!

File: core/kernel/persistence/smoothsql/class.SmoothRdf.php [hunk 3]
```diff
@@ -107,13 +106,17 @@ public function addTripleCollection(iterable $triples)
...
-        $isInsertionSuccessful = $this->insertValues($values);
-        return $isInsertionSuccessful;
+        return $this->insertValues($values);
     }
 
     protected function insertValues(array $valuesToInsert)
     {
-        return $this->getPersistence()->insertMultiple('statements', $valuesToInsert);
+        $types = [];
+        foreach ($valuesToInsert as $value) {
```
Comment (gabrielfs7) [inline core/kernel/persistence/smoothsql/class.SmoothRdf.php:115]: - [ ] Fix the expression

$value is never used

File: core/kernel/persistence/smoothsql/class.SmoothRdf.php [hunk 4]
```diff
@@ -165,4 +168,24 @@ protected function tripleToValue(core_kernel_classes_Triple $triple)
             'epoch' => $this->getPersistence()->getPlatForm()->getNowExpression()
         ];
     }
+
+    protected function getTripleParameterTypes()
+    {
+        return [
```
Comment (gabrielfs7) [inline core/kernel/persistence/smoothsql/class.SmoothRdf.php:174]: What about move this to a constant? It is used in production code and tests?

File: core/kernel/persistence/smoothsql/class.SmoothRdf.php (hunk 3 above)
Comment (siwane) [reply core/kernel/persistence/smoothsql/class.SmoothRdf.php:115]: nope but need to loop to accumulate type foreach value

File: core/kernel/persistence/smoothsql/class.SmoothRdf.php (hunk 4 above)
Comment (siwane) [reply core/kernel/persistence/smoothsql/class.SmoothRdf.php:174]: why not

Comment (siwane) [reply core/kernel/persistence/smoothsql/class.SmoothRdf.php:174]: it is used during query to ontology so prod and maybe test

File: core/kernel/persistence/newsql/NewSqlRdf.php (hunk 1 above)
Comment (siwane) [reply core/kernel/persistence/newsql/NewSqlRdf.php:106]: good catch

**Comment by gabrielfs7** [comment]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [ ] Pull request title and description are

**Comment by siwane** [comment]
@gabrielfs7 same for me, we should create a ticket for that
@boajer the current dbal version is enough for our needs, if it is required we are able to jump

**Comment by mike-ionut-mihai-sandu-tao** [comment]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful

**Pull request merged**